app/clients/ethereum/ethereum.go

```diff
@@ -131,3 +131,35 @@ func (ec *Client) waitForTransactionReceipt(hash common.Hash) (txReceipt *types.
 
 	return ec.Client.TransactionReceipt(context.Background(), hash)
 }
+
+func (ec *Client) WaitBlocks(transactionHash string, blockNumber uint64) error {
+	numberOfBlocks := ec.config.WaitingBlocks
+	if numberOfBlocks < 1 {
+		return errors.New("numberOfBlocks should be a positive number")
+	}
+
+	target := new(big.Int).Add(new(big.Int).SetUint64(blockNumber), new(big.Int).SetUint64(numberOfBlocks))
+	for {
+		currentBlockNumber, err := ec.BlockNumber(context.Background())
+		if err != nil {
+			return err
+		}
+		currentBlockNumberBn := new(big.Int).SetUint64(currentBlockNumber)
+
+		if target.Cmp(currentBlockNumberBn) <= 0 {
+			_, isPending, err := ec.Client.TransactionByHash(context.Background(), common.HexToHash(transactionHash))
+			if err != nil {
+				return err
+			}
+			if isPending {
+				err := ec.WaitBlocks(transactionHash, target.Uint64())
```

Comment:
I am not entirely sure about this one. If its pending, we must `waitForTheTransactionReceipt` -> meaning that the transaction is mined and then call `WaitBlocks` with a new `target` which is the `new transaction receipt` block